code/datalab-app/src/api/api.js

```diff
@@ -0,0 +1,30 @@
+import express from 'express';
+import bodyParser from 'body-parser';
+import { graphqlExpress, graphiqlExpress } from 'graphql-server-express';
+import http from 'http';
+import schema from './schema/index';
+
+const port = process.env.PORT || 8000;
+
+const api = graphqlExpress({ schema });
+const graphiql = graphiqlExpress({ endpointURL: '/api' });
+const app = express();
+const server = http.createServer(app);
+
+app.use((req, res, next) => {
+  res.header('Access-Control-Allow-Origin', '*');
+  res.header('Access-Control-Allow-Headers', 'Origin, X-Requested-With, Content-Type, Accept');
+  next();
+});
+
+app.options('/*', (req, res) => {
+  res.send(204);
+});
+
+app.use('/api', bodyParser.json(), api);
```

Comment:
Interesting that ```bodyParser.json()``` is only attached to this one route. Is that because it mustn't apply to the GraphiQL route?